### PR TITLE
TS - ColorType clean up

### PR DESCRIPTION
**Pull request by britt6612**
#### What does this PR do?
This PR starts the clean up of the over use of type definitions 
#### Where should the reviewer start?
js/utils/index.d.ts
#### What testing has been done on this PR?
manually tested by branching out to a typescript project
#### What are the relevant issues?
relevant issue #3237
#### Do the grommet docs need to be updated?
no
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: src/js/utils/index.d.ts [hunk 1]
```diff
@@ -27,9 +27,13 @@ export interface DeepMerge {
...
 
+
 declare const isObject: (item:any) => boolean;
 declare const deepFreeze: DeepFreeze;
 declare const deepMerge: DeepMerge;
 declare const removeUndefined: <T extends object>(obj: T) => NonUndefinedProps<T>;
 
 export {isObject, deepFreeze, deepMerge, removeUndefined};
+
+//Extracting types for common properties among components
```
Comment (ShimiSun) [inline src/js/utils/index.d.ts]:
```suggestion
// Extracting types for common properties among components
```

**Comment by britt6612** [comment]
@ShimiSun fixed merge conflicts and cleaned up extra lines

**Comment by oorestisime** [comment]
Was going to merge this but with the other merge it now has conflicts (due to imports i guess). Looks great!

**Comment by britt6612** [comment]
@oorestisime just cleaned up imports should be fine after checks are passed!

**Pull request merged**